replica/database.cc

```diff
@@ -742,6 +756,22 @@ do_parse_schema_tables(distributed<service::storage_proxy>& proxy, const sstring
     });
 }
 
+future<> database::set_in_critical_disk_utilization_mode(sharded<database>& sharded_db, bool enabled)
+{
+    return sharded_db.invoke_on_all([enabled] (replica::database& db) {
+        dblog.info("Setting critical disk utilization mode: {}", enabled);
+        if (enabled) {
+            ++db._critical_disk_utilization_mode_count;
+        } else if (db._critical_disk_utilization_mode_count > 0) {
```

Comment:
This `if` is fishy, how can the increments and decrements be unmatched?

Reply:
Then please drop the special handling, working around an impossible bug reduces confidence in the code.